### PR TITLE
Add client https method

### PR DESCRIPTION
**Pull request by shzcore**

### CODE REVIEW

File: hubble-be/src/main/java/com/baidu/hugegraph/options/HubbleOptions.java [hunk 1]
```diff
@@ -190,4 +191,32 @@ public static synchronized HubbleOptions instance() {
...
+
+    public static final ConfigOption<String> CLIENT_TRUSTSTORE_PASSWORD =
+            new ConfigOption<>(
+                    "ssl.client_truststore_password",
+                    "The password of the path of the client truststore " +
+                    "file used when the https protocol is enabled",
+                    null,
+                    ""
+            );
+
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/options/HubbleOptions.java:221]: remove empty line

File: hubble-be/src/main/java/com/baidu/hugegraph/util/HugeClientUtil.java [hunk 2]
```diff
@@ -39,17 +39,26 @@ public static HugeClient tryConnect(GraphConnection connection) {
         String username = connection.getUsername();
         String password = connection.getPassword();
         int timeout = connection.getTimeout();
+        String protocol = (connection.getProtocol() == null) ? "http" :
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/util/HugeClientUtil.java:42]: define const var

File: hubble-be/src/main/java/com/baidu/hugegraph/util/HugeClientUtil.java [hunk 3]
```diff
@@ -39,17 +39,26 @@ public static HugeClient tryConnect(GraphConnection connection) {
...
 
         String url = UriComponentsBuilder.newInstance()
                                          .scheme("http")
                                          .host(host).port(port)
                                          .toUriString();
         HugeClient client;
         try {
-            if (username != null) {
-                client = new HugeClient(url, graph, username, password, timeout);
+            if (protocol.equals("https")) {
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/util/HugeClientUtil.java:53]: define const var

File: hubble-be/src/main/java/com/baidu/hugegraph/util/HugeClientUtil.java [hunk 4]
```diff
@@ -39,17 +39,26 @@ public static HugeClient tryConnect(GraphConnection connection) {
...
+                client = new HugeClient(url, graph, username, password, timeout,
+                                        protocol, trustStoreFile, trustStorePassword);
             } else {
-                client = new HugeClient(url, graph, timeout);
+                if (username != null) {
+                    client = new HugeClient(url, graph, username, password, timeout);
+                } else {
+                    client = new HugeClient(url, graph, timeout);
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/util/HugeClientUtil.java:60]: use builder

File: hubble-be/src/main/java/com/baidu/hugegraph/options/HubbleOptions.java [hunk 5]
```diff
@@ -190,4 +191,31 @@ public static synchronized HubbleOptions instance() {
                     positiveInt(),
                     10 * Bytes.GB
             );
+
+    public static final ConfigOption<String> SERVER_PROTOCOL =
+            new ConfigOption<>(
+                    "server.protocol",
+                    "The Protocol of HugeGraph-Server,allowed values are: " +
```
Comment (Linary) [inline hubble-be/src/main/java/com/baidu/hugegraph/options/HubbleOptions.java:198]: protocal, `HugeGraph-Server,allowed -> HugeGraph-Server, allowed`

File: hubble-be/src/main/java/com/baidu/hugegraph/options/HubbleOptions.java [hunk 6]
```diff
@@ -190,4 +191,31 @@ public static synchronized HubbleOptions instance() {
...
+                    "The path of client truststore file used when https " +
+                    "protocol is enabled",
+                    null,
+                    ""
+            );
+
+    public static final ConfigOption<String> CLIENT_TRUSTSTORE_PASSWORD =
+            new ConfigOption<>(
+                    "ssl.client_truststore_password",
+                    "The password of the path of the client truststore " +
```
Comment (Linary) [inline hubble-be/src/main/java/com/baidu/hugegraph/options/HubbleOptions.java:216]: remove path

File: hubble-be/src/main/java/com/baidu/hugegraph/service/load/LoadTaskService.java [hunk 7]
```diff
@@ -284,6 +290,7 @@ public void updateLoadTaskProgress() {
 
     private LoadTask buildLoadTask(GraphConnection connection,
                                    FileMapping fileMapping) {
+        connection = sslService.configSSL(this.config, connection);
```
Comment (Linary) [inline hubble-be/src/main/java/com/baidu/hugegraph/service/load/LoadTaskService.java:293]: `sslService.configSSL` should called at outside, this method should just generate LoadOptions  and LoadOptions

File: hubble-be/src/main/java/com/baidu/hugegraph/util/HugeClientUtil.java [hunk 8]
```diff
@@ -32,25 +32,37 @@
...
+        String trustStoreFile = connection.getTrustStoreFile();
+        String trustStorePassword = connection.getTrustStorePassword();
 
         String url = UriComponentsBuilder.newInstance()
-                                         .scheme("http")
+                                         .scheme(DEFAULT_PROTOCOL)
                                          .host(host).port(port)
                                          .toUriString();
         HugeClient client;
+        if (username == null) {
```
Comment (Linary) [inline hubble-be/src/main/java/com/baidu/hugegraph/util/HugeClientUtil.java:55]: move if {} block before `HugeClient client;`

File: hubble-fe/yarn.lock [hunk 9]
```diff
@@ -275,11 +275,14 @@
   resolved "http://registry.npm.baidu-int.com/@babel%2fhelper-validator-identifier/-/helper-validator-identifier-7.10.1.tgz#5770b0c1a826c4f53f5ede5e153163e0318e94b5"
   integrity sha1-V3CwwagmxPU/Xt5eFTFj4DGOlLU=
 
+<<<<<<< HEAD
```
Comment (Linary) [inline hubble-fe/yarn.lock:278]: don't commit this file

File: hubble-be/src/main/java/com/baidu/hugegraph/util/HugeClientUtil.java [hunk 10]
```diff
@@ -32,34 +32,37 @@
...
+
     public static HugeClient tryConnect(GraphConnection connection) {
         String graph = connection.getGraph();
         String host = connection.getHost();
         Integer port = connection.getPort();
         String username = connection.getUsername();
         String password = connection.getPassword();
         int timeout = connection.getTimeout();
-        String protocol = (connection.getProtocol() == null) ? "http" :
+        String protocol = (connection.getProtocol() == null) ?
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/util/HugeClientUtil.java:44]: remove "("

**Pull request merged**